doc/SAI-Proposal-15-UOM.md

```diff
@@ -0,0 +1,229 @@
+        SAI_VLAN_MEMBER_ATTR_PORT_ID,
+ 
+        /** VLAN tagging mode [sai_vlan_tagging_mode_t] (CREATE_AND_SET)
+         * (default to SAI_VLAN_PORT_UNTAGGED) */
+        SAI_VLAN_MEMBER_ATTR_TAGGING_MODE,
+ 
+        /** custom range base value */
+        SAI_VLAN_MEMBER_ATTR_CUSTOM_RANGE_BASE  = 0x10000000
+ 
+    } sai_vlan_member_attr_t;
+
+    typedef sai_status_t (*sai_create_vlan_member_fn)(
+        _Out_ sai_object_id_t* vlan_member_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list
+        );
+
+    typedef sai_status_t (*sai_remove_vlan_member_fn)(
+        _In_ sai_object_id_t vlan_member_id
+       );
+
+    typedef sai_status_t (*sai_set_vlan_member_attribute_fn)(
+       _In_ sai_object_id_t vlan_member_id,
+       _In_ const sai_attribute_t *attr
+       );
+ 
+    typedef sai_status_t (*sai_get_vlan_member_attribute_fn)(
+       _In_ sai_object_id_t vlan_member_id,
+       _In_ const uint32_t attr_count,
+       _Inout_ sai_attribute_t *attr_list
+       );
+
+In the VLAN object, add SAI_VLAN_ATTR_MEMBER_LIST to query the vlan members. remove SAI_VLAN_ATTR_PORT_LIST. 
+ 
+    typedef enum _sai_vlan_attr_t
+    {
+        /** READ-ONLY */
+
+	    /** List of vlan member ports in a VLAN [sai_object_list_t] */
+	    SAI_VLAN_ATTR_MEMBER_LIST,
+    }
+
+### Next hop group member
+
+Add next hop group member object and create/remove/get/set API.
+
+    typedef enum _sai_next_hop_group_member_attr_t {
+ 
+        /** READ_WRITE */
+ 
+        /** NEXT HOP GROUP ID [sai_vlan_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+        SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID,
+ 
+        /** logical port ID [sai_object_id_t] (MANDATORY_ON_CREATE|CREATE_ONLY) */
+        SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_ID,
+ 
+        /** NEXT HOP GROUP Weights (CREATE_AND_SET)
+         * (default to 1) */
+        SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT,
+ 
+        /** custom range base value */
+        SAI_NEXT_HOP_GROUP_MEMBER_ATTR_CUSTOM_RANGE_BASE  = 0x10000000
+ 
+    } sai_next_hop_group_member_attr_t;
+
+
+    typedef sai_status_t (*sai_create_next_hop_group_member_fn)(
+        _Out_ sai_object_id_t* next_hop_group_member_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list
+        );
+
+    typedef sai_status_t (*sai_remove_next_hop_group_member_fn)(
+        _In_ sai_object_id_t next_hop_group_member_id
+       );
+
+    typedef sai_status_t (*sai_set_next_hop_group_member_attribute_fn)(
+       _In_ sai_object_id_t next_hop_group_member_id,
+       _In_ const sai_attribute_t *attr
+       );
+ 
+    typedef sai_status_t (*sai_get_next_hop_group_member_attribute_fn)(
+       _In_ sai_object_id_t next_hop_group_member_id,
+       _In_ const uint32_t attr_count,
+       _Inout_ sai_attribute_t *attr_list
+       );
+
+In the nexthopgroup object, remove SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_LIST attribute. 

```

Comment:
add new RO SAI_NEXT_HOP_GROUP_ATTR_NEXT_HOP_MEMBER_LIST attribute in pull request https://github.com/opencomputeproject/SAI/pull/145